### DISCUSSION TITLE
Disabling Safety Checks for RL Training with SUMO in Eclipse MOSAIC

### DISCUSSION THREAD

**Discussion by yibork** [eclipse/mosaic#434] [Q&A] [answered]
Hello Community,
I'm working on an RL-based project using Eclipse MOSAIC with SUMO as the simulator, and I’m looking to disable specific safety checks that may interfere with reinforcement learning exploration. Specifically, I want to let some vehicles behave freely without adhering to collision avoidance and speed constraints to help the RL agent learn a broader range of driving strategies, including potentially risky maneuvers.
I’ve found that SUMO allows control over speed gain changes (bit5, bit4 settings), but I’m not certain if this is sufficient to remove all the safety constraints needed for RL learning. Has anyone worked with a setup where certain safety features were removed or adjusted for similar purposes? Any insights into adjusting MOSAIC settings to enable RL control while having other vehicles continue with SUMO’s standard random behaviors would be greatly appreciated.
Thanks in advance for any suggestions or guidance on achieving this in MOSAIC.
Best,

**Comment by kschrab** [comment, marked as answer]
@yibork 

I just did a few tests and was successfully able to let two vehicles collide on an intersection, if that's it what you want to do?

![sumo_collision](https://github.com/user-attachments/assets/68d861f4-29d9-49ed-9f07-eb0d9497507e)

I achieved that by mapping a simple application to both vehicles which initially sets the SpeedMode to `AGGRESSIVE`:

```java
@Override
public void onStartup() {
    getOs().requestVehicleParametersUpdate().changeSpeedMode(SpeedMode.AGGRESSIVE).apply();
}
```

I also added `--collision.action warn --collision.check-junctions true` to the `sumo_config.json`:

```json
{
    "updateInterval" : 100,
    "additionalSumoParameters": "--time-to-teleport 10 --seed 100000 --collision.action warn --collision.check-junctions true"
}
```

However, I found that in our current implementation we do not set bit 5 yet, presumably because it was just recently added by SUMO? So in order to make that work, you have to change the `SumoSpeedMode` class to set this bit for the mode `AGGRESSIVE`. We probably add this by default or adding a new SpeedMode with all checks disabled in a future commit.

In SumoSpeedMode.java line 33-36:
```diff
   case AGGRESSIVE:
       mode.setRegardMaximumAcceleration(true);
       mode.setRegardMaximumDeceleration(true);
+      mode.bitset.set(5);
       break;
```

**Comment by yibork** [comment]
Hello,
thanks for your answer it worked for me I had also to change lanechangemode to aggressive and I started having collisions.
